01-data-structures/02-stacks-and-queues/mystack/mystack.rb

```diff
@@ -3,15 +3,35 @@ class MyStack
 
   def initialize
     @stack = Array.new
+    @stack_position = -1
     self.top = nil
   end
 
+
+
   def push(item)
+    @stack_position += 1
+    @stack[@stack_position] = item
+    # reflect the new top of the stack
+    self.top = @stack[@stack_position]
   end
 
   def pop
+    temp = @stack[@stack_position]
```

Comment:
Given the way you get the _top_ element, here, you have defeated the purpose of `self.top` altogether. Your implementation is fine, but your maintenance of the `self.top` reference is wasteful. I would choose to use it or scrap it.